src/deriveAddress.c

```diff
@@ -68,11 +68,11 @@ static void io_exchange_address()
 {
```

Comment:
Can we refactor out io_exchange? In the first iteration, let's get the addressBuffer and addressSize as parameters.
Later let's rename it to `io_respond_with_address` or something along these lines